app/controllers/transcribe_controller.rb

```diff
@@ -123,113 +123,126 @@ def needs_review
-          if @page.field_based && !table_cells.empty?
-            @page.replace_table_cells(table_cells)
-          end
+          @page.replace_table_cells(table_cells) if @page.field_based && !table_cells.empty?
 
           log_transcript_success
           flash[:notice] = t('.saved_notice')
+
           if @page.work.ocr_correction
             record_deed(DeedType::OCR_CORRECTED)
-          else
-            if @page.source_text_previously_changed?
-              record_transcription_deed
-            end
+          elsif @page.source_text_previously_changed?
+            record_transcription_deed
           end
-          #don't reset subjects if they're disabled
-          unless @page.collection.subjects_disabled || (@page.source_text.include?("[[") == false)
+
+          # don't reset subjects if they're disabled
+          unless @page.collection.subjects_disabled || (@page.source_text.include?('[[') == false)
             # use the new links to blank the graphs
             @page.clear_article_graphs
 
             new_link_count = @page.page_article_links.where(text_type: 'transcription').count
             logger.debug("DEBUG old_link_count=#{old_link_count}, new_link_count=#{new_link_count}")
-            if old_link_count == 0 && new_link_count > 0
-              record_deed(DeedType::PAGE_INDEXED)
-            end
-            if new_link_count > 0 && @page.status != Page::STATUS_NEEDS_REVIEW && @page.status != Page::STATUS_INCOMPLETE
+
+            record_deed(DeedType::PAGE_INDEXED) if old_link_count.zero? && new_link_count.positive?
+
+            if new_link_count.positive? &&
+               @page.status != Page::STATUS_NEEDS_REVIEW &&
+               @page.status != Page::STATUS_INCOMPLETE
+
               @page.update_columns(status: Page::STATUS_INDEXED)
             end
           end
-          @work.work_statistic.recalculate({type: @page.status}) if @work.work_statistic
-          @page.submit_background_processes("transcription")
 
-          #if this is a guest user, force them to sign up after three saves
+          @work.work_statistic&.recalculate({ type: @page.status })
+          @page.submit_background_processes('transcription')
+
+          # if this is a guest user, force them to sign up after three saves
           if current_user.guest?
             deeds = Deed.where(user_id: current_user.id).where(deed_type: DeedType.edited_and_transcribed_pages).count
             if deeds < GUEST_DEED_COUNT
               flash[:notice] = t('.you_may_save_notice', guest_deed_count: GUEST_DEED_COUNT)
             else
-              session[:user_return_to]=collection_transcribe_page_path(@collection.owner, @collection, @work, @page.id)
-              redirect_to new_user_registration_path, :resource => current_user
+              session[:user_return_to] = collection_transcribe_page_path(@collection.owner,
+                                                                         @collection, @work, @page.id)
```

Comment:
If parameters in a method call don't all fit on one line, let's put each of them on their own line.